packages/web-components/src/components/va-pagination/va-pagination.tsx

```diff
@@ -125,66 +167,81 @@ export class VaPagination {
       unbounded,
     } = this;
 
-    // radius is half the length of the visible pages
-    // use as a reference from the selected page to find
-    // start and end of the pages to render
+    // Radius is half the length of the visible pages
     const radius = Math.floor(maxPageListLength / 2);
 
-    //if the unbounded flag is set we don't include the last page
+    // If the unbounded flag is set we don't include the last page
     const unboundedChar = unbounded ? 0 : 1;
 
-    let start;
-    let end;
+    let { isMobileViewport, maxPageListLengthState, SHOW_ALL_PAGES } = this;
+    let start: number;
+    let end: number;
 
+    // The smallest breakpoint can show at most 6 pages, including ellipses
+    if (isMobileViewport) maxPageListLengthState = SHOW_ALL_PAGES;
+
+    // Use cases
```

Comment:
I lightly refactored the `pageNumbers()` function to use `if, else if, else` logic. I explored using a switch case but didn't feel it was going to scale with multiple props needed to derive the final arrays.